### ISSUE TITLE
XPath matching

### ISSUE THREAD

**Issue by beporter** [mawenbao/gofeed#12]
This is [already mentioned in the README](https://github.com/mawenbao/gofeed#more-functions-on-the-todo-list), but I figured it'd make sense to have it as an actual Issue.

It would extremely helpful to be able to match items using XPath instead of string matching or regular expressions. I realize this is somewhat difficult due to the limited XPath availability in Golang right now of course.

**Comment by mawenbao** [comment]
Yes, XPath is more convenient than regex especially when the target html structure changes all the time.  As far as I know, there is already a Go library called [xpath](http://godoc.org/launchpad.net/xmlpath). I will look into this when I have the time.
